### PR TITLE
Fix chat encoding

### PR DESCRIPTION
**Pull request by stenliis**
Fix for #234.

Tested mainly with Czech keyboard, that contains characters that were also broken (ěšěčšřčžřýáíé...).
Done little testing with Russian keyboard and look to me fixed. 
Encoding-cz
Encoding-ru

### CODE REVIEW

File: Code/client/Services/Generic/OverlayClient.cpp [hunk 1]
```diff
@@ -110,6 +110,7 @@ void OverlayClient::ProcessChatMessage(CefRefPtr<CefListValue> aEventArgs)
         messageRequest.MessageType = static_cast<ChatMessageType>(aEventArgs->GetInt(0));
         messageRequest.ChatMessage = contents;
 
+        // Logging of unicode character is broken and fixing it is not realistic
```
Comment (Force67) [inline Code/client/Services/Generic/OverlayClient.cpp:113]: Its possible by logging a wide string to spdlog.

File: Code/client/Services/Generic/InputService.cpp [hunk 2]
```diff
@@ -414,7 +433,20 @@ LRESULT CALLBACK InputService::WndProc(HWND hwnd, UINT uMsg, WPARAM wParam, LPAR
     }
     else if (uMsg == WM_CHAR)
     {
-        ProcessKeyboard(static_cast<uint16_t>(wParam), (lParam >> 16) & 0xFF, KEYEVENT_CHAR, false, false);
+        uint16_t scancode = (lParam >> 16) & 0xFF;
+        uint16_t virtualKey = static_cast<uint16_t>(wParam);
+        if (!IsWindowUnicode(hwnd))
+        {
+            wchar_t wch;
+            ::MultiByteToWideChar(s_currentACP, MB_PRECOMPOSED, (char*) &virtualKey, 2, &wch, sizeof(wchar_t));
```
Comment (Force67) [inline Code/client/Services/Generic/InputService.cpp:441]: No c style casts

**Comment by Force67** [comment]
Willing to fix this sometime?

**Comment by stenliis** [comment]
> Willing to fix this sometime?

I tried to fix it and failed.

I have issue with SPDLOG_WCHAR_TO_UTF8_SUPPORT, it fail to compile.
The compiler reports multiple error of this kind: 
C:\Users\stenliis\AppData\Local\.xmake\packages\s\spdlog\v1.10.0\6bf6da47960848619cd072b319304dad\include\spdlog/spdlog.h(192): error C7568: argument list missing after assumed function template 'wformat_string_t'

I did not manage to find the exact cause of this error.

So feel free to try to fix it.

**Comment by stenliis** [comment]
encoding in file and console fixed 
log-fix

**Pull request merged**